cmd/collector/app/server/http.go

```diff
@@ -40,11 +39,9 @@ func StartHTTPServer(params *HTTPServerParams) (*http.Server, error) {
 	if err != nil {
 		return nil, err
 	}
-	errorLog, _ := zap.NewStdLogAt(params.Logger, zapcore.ErrorLevel)
-	server, err := params.ToServer(context.Background(), nil, telemetry.NoopSettings().ToOtelComponent(),
-		nil)
-
-	server.ErrorLog = errorLog
+	settings := telemetry.NoopSettings().ToOtelComponent()
+	settings.Logger = params.Logger
+	server, err := params.ToServer(context.Background(), nil, settings, nil)
```

Comment:
@yurishkuro we can make this change due to https://github.com/open-telemetry/opentelemetry-collector/issues/11820 being completed